File: daily_python/1009_Complement_of_Base_10_Integer.py
import os
import sys
from typing import List

class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        num = n
        l = 0
        while num > 0:
            num //= 2
            l += 1
        return (1 << l) - 1 - n

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            n = int(lines[0].split('=')[1].strip()[:-1])
            dataset.append(n)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for n in dataset:
        results.append(solution.bitwiseComplement(n))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/1009_Complement_of_Base_10_Integer.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int bitwiseComplement(int n) {
        if (n == 0) return 1;
        int l = 0;
        int num = n;
        while(num){
            num >>= 1;
            l++;
        }
        return (1 << l) - 1 - n;
    }
};

int main(){
    int n = 5;
    // int n = 7;
    // int n = 10;
    // int n = 0;
    Solution S;
    cout << S.bitwiseComplement(n) << endl;
    return 0;
}